File: Test_Reminder_Flow_Monolegal_Backend/src/domain/value_objects/invoice_status.py
from enum import Enum


class InvoiceStatus(str, Enum):
    PENDIENTE = "pendiente"
    PRIMER_RECORDATORIO = "primerrecordatorio"
    SEGUNDO_RECORDATORIO = "segundorecordatorio"
    DESACTIVADO = "desactivado"

    @classmethod
    def get_next_status(cls, current_status: "InvoiceStatus") -> "InvoiceStatus":
        transitions = {
            cls.PENDIENTE: cls.PRIMER_RECORDATORIO,
            cls.PRIMER_RECORDATORIO: cls.SEGUNDO_RECORDATORIO,
            cls.SEGUNDO_RECORDATORIO: cls.DESACTIVADO,
        }
        return transitions.get(current_status, current_status)

    @classmethod
    def is_actionable(cls, status: "InvoiceStatus") -> bool:
        return status in [cls.PENDIENTE, cls.PRIMER_RECORDATORIO, cls.SEGUNDO_RECORDATORIO]

File: Test_Reminder_Flow_Monolegal_Backend_New/src/ReminderFlow.Domain/ValueObjects/InvoiceStatus.cs
namespace ReminderFlow.Domain.ValueObjects;

public enum InvoiceStatus
{
    Pendiente,
    PrimerRecordatorio,
    SegundoRecordatorio,
    Desactivado
}

public static class InvoiceStatusExtensions
{
    public static InvoiceStatus GetNextStatus(this InvoiceStatus status)
    {
        return status switch
        {
            InvoiceStatus.Pendiente => InvoiceStatus.PrimerRecordatorio,
            InvoiceStatus.PrimerRecordatorio => InvoiceStatus.SegundoRecordatorio,
            InvoiceStatus.SegundoRecordatorio => InvoiceStatus.Desactivado,
            InvoiceStatus.Desactivado => InvoiceStatus.Desactivado,
            _ => status
        };
    }

    public static bool IsActionable(this InvoiceStatus status)
    {
        return status is InvoiceStatus.Pendiente 
            or InvoiceStatus.PrimerRecordatorio 
            or InvoiceStatus.SegundoRecordatorio;
    }

    public static string ToDbString(this InvoiceStatus status)
    {
        return status switch
        {
            InvoiceStatus.Pendiente => "pendiente",
            InvoiceStatus.PrimerRecordatorio => "primerrecordatorio",
            InvoiceStatus.SegundoRecordatorio => "segundorecordatorio",
            InvoiceStatus.Desactivado => "desactivado",
            _ => "pendiente"
        };
    }

    public static InvoiceStatus FromDbString(string status)
    {
        return status.ToLower() switch
        {
            "pendiente" => InvoiceStatus.Pendiente,
            "primerrecordatorio" => InvoiceStatus.PrimerRecordatorio,
            "segundorecordatorio" => InvoiceStatus.SegundoRecordatorio,
            "desactivado" => InvoiceStatus.Desactivado,
            _ => InvoiceStatus.Pendiente
        };
    }
}
